### PR TITLE
Make /lgtm and /approve also act the same as /ok-to-test

### PR DESCRIPTION
**Pull request by intUnderflow**
If a contributor gives a /lgtm or /approve on a PR there is a strong implication they believe the PR is also ok to test

It is very easy for contributors to miss the fact that they need to also provide /ok-to-test if they want the tests to run and the PR to become mergeable, especially as they only need to do this for folks who are not part of the org

This PR makes prow treat /lgtm and /approve the same as /ok-to-test, making the PR trusted and kicking off tests

### CODE REVIEW

**Comment by dims** [comment]
should i do /ok-to-test or /lgtm here? :)


/ok-to-test

**Review by cjwagner — CHANGES_REQUESTED** [review]
We cannot merge this, this would be very unsafe.

> It is very easy for contributors to miss the fact that they need to also provide /ok-to-test if they want the tests to run and the PR to become mergeable, especially as they only need to do this for folks who are not part of the org

This is not true. `/ok-to-test` is not required to run tests or merge PRs, it just indicates that the PR author can run tests on the PR themselves even if they are not a member.

An LGTM or approve from a trusted member on an untrusted PR will still cause tests to trigger, Tide to retest, and the PR to eventually merge. It just prevents the PR author from changing the PR and running presubmits again themselves (potentially with something malicious).

> This PR makes prow treat /lgtm and /approve the same as /ok-to-test, making the PR trusted and kicking off tests

This would make it impossible to merge a PR without granting the PR author permission to run presubmits on arbitrary source changes which would be a major security regression.

**Comment by intUnderflow** [comment]
Hey @cjwagner, thanks for the background :)

Just to clarify on:
> This would make it impossible to merge a PR without granting the PR author permission to run presubmits on arbitrary source changes which would be a major security regression.

Is this a use case that people depend on right now? It seems surprising to me that a PR would be given LGTM but the author would still not be trusted to spam the presubmit system but perhaps I'm missing something

**Comment by cjwagner** [comment]
> Is this a use case that people depend on right now? It seems surprising to me that a PR would be given LGTM but the author would still not be trusted to spam the presubmit system but perhaps I'm missing something?

Yes it is actually common case for any first time contributors. In this case, since we have never seen any contributions from the GitHub account we don't trust it to to run arbitrarily many presubmits on arbitrary source changes (the latter being the most significant concern). A reviewer should LGTM/approve as appropriate once they've reviewed the PR to start a round of testing and allow the PR to eventually merge. No additional human interactions are necessary to merge the PR after LGTM+Approve assuming the presubmits all pass.

If the reviewer had also granted `/ok-to-test` after their review and the new, untrusted GitHub account turned out to be malicious, they would then be able to hold the PR and include malicious presubmit logic before triggering as many presubmit runs as they'd like. Malicious presumits might just steal/waste computing resources, but they could also potentially steal credentials or otherwise compromise our services.

**Comment by SergeyKanzhelev** [comment]
> An LGTM or approve from a trusted member on an untrusted PR will still cause tests to trigger, Tide to retest, and the PR to eventually merge. It just prevents the PR author from changing the PR and running presubmits again themselves (potentially with something malicious).

I have a feeling that this is not the common knowledge. I wonder how it works, my understanding was that the ride batches PRs. So one lgtm-ed and approved PR that makes test fail will fail the entire batch? And there is no message sent back to the  "bad" PR? I really like this level of protection of the project, but maybe this scenario needs to be thought-thru and communicated widely.

As for this PR, it would be great to have an option to allow tests for lgtm-ed code only. Like `/ok-to-test-lgtm`.

**Comment by cjwagner** [comment]
> I have a feeling that this is not the common knowledge.

Yeah, unfortunately I think many folks see others using `/ok-to-test` and think it is required on non-member PRs.

> So one lgtm-ed and approved PR that makes test fail will fail the entire batch? And there is no message sent back to the "bad" PR?

Only PRs that have passed presubmits at some point in the past are included in a batch, so if a PR is failing tests it won't be included in the batch in the first place. If something conflicting has merged since the presubmits last passed the PR could be included in the batch and the batch could fail. However since we serially test a member of the batch we ensure that we'll make progress by merging at least one PR each time or, in the case of the now conflicting PR, the serial tests would fail causing the PR to be removed from future batches and also resulting in a comment on the "bad" PR.

> As for this PR, it would be great to have an option to allow tests for lgtm-ed code only. Like /ok-to-test-lgtm or simply allow tests on all lgtm-ed code, but not on modifications.

Those are interesting ideas. If you'd like to pursue them the first step would be running this by SIG Contribex and getting their approval.

**Pull request closed without merging**
